transport/http/outbound.go

```diff
@@ -386,29 +390,28 @@ func (o *Outbound) withCoreHeaders(req *http.Request, treq *transport.Request, t
 	return req
 }
 
-func getErrFromResponse(response *http.Response) error {
-	// TODO Behavior for 300-range status codes is undefined
+func getYARPCErrorFromResponse(response *http.Response) error {
 	contents, err := ioutil.ReadAll(response.Body)
 	if err != nil {
-		return err
+		return yarpcerrors.InternalErrorf(err.Error())
 	}
-
 	if err := response.Body.Close(); err != nil {
-		return err
-	}
-
-	// Trim the trailing newline from HTTP error messages
-	message := strings.TrimSuffix(string(contents), "\n")
-
-	if response.StatusCode >= 400 && response.StatusCode < 500 {
-		return errors.RemoteBadRequestError(message)
-	}
-
-	if response.StatusCode == http.StatusGatewayTimeout {
-		return errors.RemoteTimeoutError(message)
+		return yarpcerrors.InternalErrorf(err.Error())
+	}
+	// use the status code if we can't get a code from the headers
+	code := StatusCodeToBestCode(response.StatusCode)
+	if errorCodeText := response.Header.Get(ErrorCodeHeader); errorCodeText != "" {
+		var errorCode yarpcerrors.Code
+		// TODO: what to do with error?
+		if err := (&errorCode).UnmarshalText([]byte(errorCodeText)); err == nil {
```

Comment:
This is for `UnmarshalText`, which sets the value of `errorCode` https://godoc.org/encoding#TextUnmarshaler. You need a pointer here I think.